mypyc 0.910 - 0.931 test failures on i686 (and armel, armhf)

**Bug Report**

Some `mypyc/test/test_run.py::TestRun` tests failed during the build of the Debian package for mypy 0.910.

**To Reproduce**

Save the following as `Dockerfile` in an empty directory
``` Dockerfile
FROM debian:unstable-slim
RUN apt-get update && apt-get install -y python3-pip python3-lxml wget make
WORKDIR /tmp
RUN wget https://github.com/python/mypy/archive/refs/tags/v0.910.tar.gz && \
	tar xzf v*tar.gz && \
	rm *.tar.gz
WORKDIR /tmp/mypy-0.910
RUN python3 -m pip install -r test-requirements.txt
RUN python3 -m pytest -v -k "testSubclassSpecialize1 or testSubclassSpecialize2 or testMultiModuleSpecialize or testMultiModuleSpecialize_multi or testMultiModuleSpecialize_separate"
```

And build an i386/i686 container using [`docker buildx`](https://docs.docker.com/desktop/multi-arch/#build-and-run-multi-architecture-images): `docker buildx build --platform i386 .`

**Expected Behavior**

All tests pass

**Actual Behavior**
```
___________________________ testSubclassSpecialize1 ____________________________
[gw2] linux -- Python 3.9.7 /usr/bin/python3.9
data: /<<PKGBUILDDIR>>/mypyc/test-data/run-classes.test:357:
/<<PKGBUILDDIR>>/mypyc/test/test_run.py:133: in run_case
    self.run_case_inner(testcase)
/<<PKGBUILDDIR>>/mypyc/test/test_run.py:148: in run_case_inner
    self.run_case_step(testcase, step)
/<<PKGBUILDDIR>>/mypyc/test/test_run.py:322: in run_case_step
    assert_test_output(testcase, outlines, msg, expected)
/<<PKGBUILDDIR>>/mypyc/test/testutil.py:163: in assert_test_output
    assert_string_arrays_equal(
/<<PKGBUILDDIR>>/mypy/test/helpers.py:117: in assert_string_arrays_equal
    raise AssertionError(msg)
E   AssertionError: Invalid output (/<<PKGBUILDDIR>>/mypyc/test-data/run-classes.test, line 357)
----------------------------- Captured stdout call -----------------------------
running build_ext
building 'native' extension
creating build/temp.linux-i686-3.9
creating build/temp.linux-i686-3.9/build
i686-linux-gnu-gcc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O2 -Wall -g -fstack-protector-strong -Wformat -Werror=format-security -g -fwrapv -O2 -g -O2 -ffile-prefix-map=/<<PKGBUILDDIR>>=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -fPIC -I/<<PKGBUILDDIR>>/mypyc/lib-rt -I/usr/include/python3.9 -c build/__native.c -o build/temp.linux-i686-3.9/build/__native.o -O0 -Werror -Wno-unused-function -Wno-unused-label -Wno-unreachable-code -Wno-unused-variable -Wno-unused-command-line-argument -Wno-unknown-warning-option -Wno-unused-but-set-variable
creating build/lib.linux-i686-3.9
i686-linux-gnu-gcc -pthread -shared -Wl,-O1 -Wl,-Bsymbolic-functions -Wl,-z,relro -g -fwrapv -O2 -Wl,-z,relro -Wl,-z,now -g -O2 -ffile-prefix-map=/<<PKGBUILDDIR>>=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 build/temp.linux-i686-3.9/build/__native.o -o build/lib.linux-i686-3.9/native.cpython-39-i386-linux-gnu.so
copying build/lib.linux-i686-3.9/native.cpython-39-i386-linux-gnu.so -> 

*** Exit status: 1
----------------------------- Captured stderr call -----------------------------
Expected:
  A
  B
  11                                            (diff)
  A                                             (diff)
  B                                             (diff)
  10                                            (diff)
  B                                             (diff)
Actual:
  A
  B
  Traceback (most recent call last):            (diff)
    File "driver.py", line 7, in <module>       (diff)
      assert b.foo(o) == id(o)                  (diff)
  AssertionError                                (diff)

___________________________ testSubclassSpecialize2 ____________________________
[gw2] linux -- Python 3.9.7 /usr/bin/python3.9
data: /<<PKGBUILDDIR>>/mypyc/test-data/run-classes.test:398:
/<<PKGBUILDDIR>>/mypyc/test/test_run.py:133: in run_case
    self.run_case_inner(testcase)
/<<PKGBUILDDIR>>/mypyc/test/test_run.py:148: in run_case_inner
    self.run_case_step(testcase, step)
/<<PKGBUILDDIR>>/mypyc/test/test_run.py:322: in run_case_step
    assert_test_output(testcase, outlines, msg, expected)
/<<PKGBUILDDIR>>/mypyc/test/testutil.py:163: in assert_test_output
    assert_string_arrays_equal(
/<<PKGBUILDDIR>>/mypy/test/helpers.py:117: in assert_string_arrays_equal
    raise AssertionError(msg)
E   AssertionError: Invalid output (/<<PKGBUILDDIR>>/mypyc/test-data/run-classes.test, line 398)
----------------------------- Captured stdout call -----------------------------
running build_ext
building 'native' extension
creating build/temp.linux-i686-3.9
creating build/temp.linux-i686-3.9/build
i686-linux-gnu-gcc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O2 -Wall -g -fstack-protector-strong -Wformat -Werror=format-security -g -fwrapv -O2 -g -O2 -ffile-prefix-map=/<<PKGBUILDDIR>>=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -fPIC -I/<<PKGBUILDDIR>>/mypyc/lib-rt -I/usr/include/python3.9 -c build/__native.c -o build/temp.linux-i686-3.9/build/__native.o -O0 -Werror -Wno-unused-function -Wno-unused-label -Wno-unreachable-code -Wno-unused-variable -Wno-unused-command-line-argument -Wno-unknown-warning-option -Wno-unused-but-set-variable
creating build/lib.linux-i686-3.9
i686-linux-gnu-gcc -pthread -shared -Wl,-O1 -Wl,-Bsymbolic-functions -Wl,-z,relro -g -fwrapv -O2 -Wl,-z,relro -Wl,-z,now -g -O2 -ffile-prefix-map=/<<PKGBUILDDIR>>=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 build/temp.linux-i686-3.9/build/__native.o -o build/lib.linux-i686-3.9/native.cpython-39-i386-linux-gnu.so
copying build/lib.linux-i686-3.9/native.cpython-39-i386-linux-gnu.so -> 

*** Exit status: 1
----------------------------- Captured stderr call -----------------------------
Expected:
  A
  B
  C
  A                                             (diff)
  B                                             (diff)
  C                                             (diff)
  B                                             (diff)
  C                                             (diff)
  C                                             (diff)
Actual:
  A
  B
  C
  Traceback (most recent call last):            (diff)
    File "driver.py", line 9, in <module>       (diff)
      assert c.foo(o) == id(o)                  (diff)
  AssertionError                                (diff)

__________________________ testMultiModuleSpecialize ___________________________
[gw3] linux -- Python 3.9.7 /usr/bin/python3.9
data: /<<PKGBUILDDIR>>/mypyc/test-data/run-multimodule.test:186:
/<<PKGBUILDDIR>>/mypyc/test/test_run.py:133: in run_case
    self.run_case_inner(testcase)
/<<PKGBUILDDIR>>/mypyc/test/test_run.py:148: in run_case_inner
    self.run_case_step(testcase, step)
/<<PKGBUILDDIR>>/mypyc/test/test_run.py:322: in run_case_step
    assert_test_output(testcase, outlines, msg, expected)
/<<PKGBUILDDIR>>/mypyc/test/testutil.py:163: in assert_test_output
    assert_string_arrays_equal(
/<<PKGBUILDDIR>>/mypy/test/helpers.py:117: in assert_string_arrays_equal
    raise AssertionError(msg)
E   AssertionError: Invalid output (/<<PKGBUILDDIR>>/mypyc/test-data/run-multimodule.test, line 186)
----------------------------- Captured stdout call -----------------------------
running build_ext
building '3ef2a66b8671abc76f6e__mypyc' extension
creating build/temp.linux-i686-3.9
creating build/temp.linux-i686-3.9/build
i686-linux-gnu-gcc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O2 -Wall -g -fstack-protector-strong -Wformat -Werror=format-security -g -fwrapv -O2 -g -O2 -ffile-prefix-map=/<<PKGBUILDDIR>>=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -fPIC -I/<<PKGBUILDDIR>>/mypyc/lib-rt -Ibuild -I/usr/include/python3.9 -c build/__native_3ef2a66b8671abc76f6e.c -o build/temp.linux-i686-3.9/build/__native_3ef2a66b8671abc76f6e.o -O0 -Werror -Wno-unused-function -Wno-unused-label -Wno-unreachable-code -Wno-unused-variable -Wno-unused-command-line-argument -Wno-unknown-warning-option -Wno-unused-but-set-variable
creating build/lib.linux-i686-3.9
i686-linux-gnu-gcc -pthread -shared -Wl,-O1 -Wl,-Bsymbolic-functions -Wl,-z,relro -g -fwrapv -O2 -Wl,-z,relro -Wl,-z,now -g -O2 -ffile-prefix-map=/<<PKGBUILDDIR>>=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 build/temp.linux-i686-3.9/build/__native_3ef2a66b8671abc76f6e.o -o build/lib.linux-i686-3.9/3ef2a66b8671abc76f6e__mypyc.cpython-39-i386-linux-gnu.so
building 'native' extension
i686-linux-gnu-gcc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O2 -Wall -g -fstack-protector-strong -Wformat -Werror=format-security -g -fwrapv -O2 -g -O2 -ffile-prefix-map=/<<PKGBUILDDIR>>=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -fPIC -I/usr/include/python3.9 -c build/native.c -o build/temp.linux-i686-3.9/build/native.o -O0 -Werror -Wno-unused-function -Wno-unused-label -Wno-unreachable-code -Wno-unused-variable -Wno-unused-command-line-argument -Wno-unknown-warning-option -Wno-unused-but-set-variable
i686-linux-gnu-gcc -pthread -shared -Wl,-O1 -Wl,-Bsymbolic-functions -Wl,-z,relro -g -fwrapv -O2 -Wl,-z,relro -Wl,-z,now -g -O2 -ffile-prefix-map=/<<PKGBUILDDIR>>=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 build/temp.linux-i686-3.9/build/native.o -o build/lib.linux-i686-3.9/native.cpython-39-i386-linux-gnu.so
building 'other' extension
i686-linux-gnu-gcc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O2 -Wall -g -fstack-protector-strong -Wformat -Werror=format-security -g -fwrapv -O2 -g -O2 -ffile-prefix-map=/<<PKGBUILDDIR>>=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -fPIC -I/usr/include/python3.9 -c build/other.c -o build/temp.linux-i686-3.9/build/other.o -O0 -Werror -Wno-unused-function -Wno-unused-label -Wno-unreachable-code -Wno-unused-variable -Wno-unused-command-line-argument -Wno-unknown-warning-option -Wno-unused-but-set-variable
i686-linux-gnu-gcc -pthread -shared -Wl,-O1 -Wl,-Bsymbolic-functions -Wl,-z,relro -g -fwrapv -O2 -Wl,-z,relro -Wl,-z,now -g -O2 -ffile-prefix-map=/<<PKGBUILDDIR>>=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 build/temp.linux-i686-3.9/build/other.o -o build/lib.linux-i686-3.9/other.cpython-39-i386-linux-gnu.so
copying build/lib.linux-i686-3.9/3ef2a66b8671abc76f6e__mypyc.cpython-39-i386-linux-gnu.so -> 
copying build/lib.linux-i686-3.9/native.cpython-39-i386-linux-gnu.so -> 
copying build/lib.linux-i686-3.9/other.cpython-39-i386-linux-gnu.so -> 

*** Exit status: 1
----------------------------- Captured stderr call -----------------------------
Expected:
  1
  2
  1                                             (diff)
  2                                             (diff)
Actual:
  1
  2
  Traceback (most recent call last):            (diff)
    File "driver.py", line 8, in <module>       (diff)
      assert b.foo(o) == id(o)                  (diff)
  AssertionError                                (diff)

_______________________ testMultiModuleSpecialize_multi ________________________
[gw0] linux -- Python 3.9.7 /usr/bin/python3.9
data: /<<PKGBUILDDIR>>/mypyc/test-data/run-multimodule.test:186:
/<<PKGBUILDDIR>>/mypyc/test/test_run.py:133: in run_case
    self.run_case_inner(testcase)
/<<PKGBUILDDIR>>/mypyc/test/test_run.py:148: in run_case_inner
    self.run_case_step(testcase, step)
/<<PKGBUILDDIR>>/mypyc/test/test_run.py:322: in run_case_step
    assert_test_output(testcase, outlines, msg, expected)
/<<PKGBUILDDIR>>/mypyc/test/testutil.py:163: in assert_test_output
    assert_string_arrays_equal(
/<<PKGBUILDDIR>>/mypy/test/helpers.py:117: in assert_string_arrays_equal
    raise AssertionError(msg)
E   AssertionError: Invalid output (/<<PKGBUILDDIR>>/mypyc/test-data/run-multimodule.test, line 186)
----------------------------- Captured stdout call -----------------------------
running build_ext
building '3ef2a66b8671abc76f6e__mypyc' extension
creating build/temp.linux-i686-3.9
creating build/temp.linux-i686-3.9/build
i686-linux-gnu-gcc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O2 -Wall -g -fstack-protector-strong -Wformat -Werror=format-security -g -fwrapv -O2 -g -O2 -ffile-prefix-map=/<<PKGBUILDDIR>>=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -fPIC -I/<<PKGBUILDDIR>>/mypyc/lib-rt -Ibuild -I/usr/include/python3.9 -c build/__native_3ef2a66b8671abc76f6e.c -o build/temp.linux-i686-3.9/build/__native_3ef2a66b8671abc76f6e.o -O0 -Werror -Wno-unused-function -Wno-unused-label -Wno-unreachable-code -Wno-unused-variable -Wno-unused-command-line-argument -Wno-unknown-warning-option -Wno-unused-but-set-variable
i686-linux-gnu-gcc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O2 -Wall -g -fstack-protector-strong -Wformat -Werror=format-security -g -fwrapv -O2 -g -O2 -ffile-prefix-map=/<<PKGBUILDDIR>>=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -fPIC -I/<<PKGBUILDDIR>>/mypyc/lib-rt -Ibuild -I/usr/include/python3.9 -c build/__native_native.c -o build/temp.linux-i686-3.9/build/__native_native.o -O0 -Werror -Wno-unused-function -Wno-unused-label -Wno-unreachable-code -Wno-unused-variable -Wno-unused-command-line-argument -Wno-unknown-warning-option -Wno-unused-but-set-variable
i686-linux-gnu-gcc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O2 -Wall -g -fstack-protector-strong -Wformat -Werror=format-security -g -fwrapv -O2 -g -O2 -ffile-prefix-map=/<<PKGBUILDDIR>>=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -fPIC -I/<<PKGBUILDDIR>>/mypyc/lib-rt -Ibuild -I/usr/include/python3.9 -c build/__native_other.c -o build/temp.linux-i686-3.9/build/__native_other.o -O0 -Werror -Wno-unused-function -Wno-unused-label -Wno-unreachable-code -Wno-unused-variable -Wno-unused-command-line-argument -Wno-unknown-warning-option -Wno-unused-but-set-variable
i686-linux-gnu-gcc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O2 -Wall -g -fstack-protector-strong -Wformat -Werror=format-security -g -fwrapv -O2 -g -O2 -ffile-prefix-map=/<<PKGBUILDDIR>>=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -fPIC -I/<<PKGBUILDDIR>>/mypyc/lib-rt -Ibuild -I/usr/include/python3.9 -c build/dict_ops.c -o build/temp.linux-i686-3.9/build/dict_ops.o -O0 -Werror -Wno-unused-function -Wno-unused-label -Wno-unreachable-code -Wno-unused-variable -Wno-unused-command-line-argument -Wno-unknown-warning-option -Wno-unused-but-set-variable
i686-linux-gnu-gcc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O2 -Wall -g -fstack-protector-strong -Wformat -Werror=format-security -g -fwrapv -O2 -g -O2 -ffile-prefix-map=/<<PKGBUILDDIR>>=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -fPIC -I/<<PKGBUILDDIR>>/mypyc/lib-rt -Ibuild -I/usr/include/python3.9 -c build/exc_ops.c -o build/temp.linux-i686-3.9/build/exc_ops.o -O0 -Werror -Wno-unused-function -Wno-unused-label -Wno-unreachable-code -Wno-unused-variable -Wno-unused-command-line-argument -Wno-unknown-warning-option -Wno-unused-but-set-variable
i686-linux-gnu-gcc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O2 -Wall -g -fstack-protector-strong -Wformat -Werror=format-security -g -fwrapv -O2 -g -O2 -ffile-prefix-map=/<<PKGBUILDDIR>>=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -fPIC -I/<<PKGBUILDDIR>>/mypyc/lib-rt -Ibuild -I/usr/include/python3.9 -c build/generic_ops.c -o build/temp.linux-i686-3.9/build/generic_ops.o -O0 -Werror -Wno-unused-function -Wno-unused-label -Wno-unreachable-code -Wno-unused-variable -Wno-unused-command-line-argument -Wno-unknown-warning-option -Wno-unused-but-set-variable
i686-linux-gnu-gcc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O2 -Wall -g -fstack-protector-strong -Wformat -Werror=format-security -g -fwrapv -O2 -g -O2 -ffile-prefix-map=/<<PKGBUILDDIR>>=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -fPIC -I/<<PKGBUILDDIR>>/mypyc/lib-rt -Ibuild -I/usr/include/python3.9 -c build/getargs.c -o build/temp.linux-i686-3.9/build/getargs.o -O0 -Werror -Wno-unused-function -Wno-unused-label -Wno-unreachable-code -Wno-unused-variable -Wno-unused-command-line-argument -Wno-unknown-warning-option -Wno-unused-but-set-variable
i686-linux-gnu-gcc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O2 -Wall -g -fstack-protector-strong -Wformat -Werror=format-security -g -fwrapv -O2 -g -O2 -ffile-prefix-map=/<<PKGBUILDDIR>>=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -fPIC -I/<<PKGBUILDDIR>>/mypyc/lib-rt -Ibuild -I/usr/include/python3.9 -c build/getargsfast.c -o build/temp.linux-i686-3.9/build/getargsfast.o -O0 -Werror -Wno-unused-function -Wno-unused-label -Wno-unreachable-code -Wno-unused-variable -Wno-unused-command-line-argument -Wno-unknown-warning-option -Wno-unused-but-set-variable
i686-linux-gnu-gcc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O2 -Wall -g -fstack-protector-strong -Wformat -Werror=format-security -g -fwrapv -O2 -g -O2 -ffile-prefix-map=/<<PKGBUILDDIR>>=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -fPIC -I/<<PKGBUILDDIR>>/mypyc/lib-rt -Ibuild -I/usr/include/python3.9 -c build/init.c -o build/temp.linux-i686-3.9/build/init.o -O0 -Werror -Wno-unused-function -Wno-unused-label -Wno-unreachable-code -Wno-unused-variable -Wno-unused-command-line-argument -Wno-unknown-warning-option -Wno-unused-but-set-variable
i686-linux-gnu-gcc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O2 -Wall -g -fstack-protector-strong -Wformat -Werror=format-security -g -fwrapv -O2 -g -O2 -ffile-prefix-map=/<<PKGBUILDDIR>>=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -fPIC -I/<<PKGBUILDDIR>>/mypyc/lib-rt -Ibuild -I/usr/include/python3.9 -c build/int_ops.c -o build/temp.linux-i686-3.9/build/int_ops.o -O0 -Werror -Wno-unused-function -Wno-unused-label -Wno-unreachable-code -Wno-unused-variable -Wno-unused-command-line-argument -Wno-unknown-warning-option -Wno-unused-but-set-variable
i686-linux-gnu-gcc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O2 -Wall -g -fstack-protector-strong -Wformat -Werror=format-security -g -fwrapv -O2 -g -O2 -ffile-prefix-map=/<<PKGBUILDDIR>>=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -fPIC -I/<<PKGBUILDDIR>>/mypyc/lib-rt -Ibuild -I/usr/include/python3.9 -c build/list_ops.c -o build/temp.linux-i686-3.9/build/list_ops.o -O0 -Werror -Wno-unused-function -Wno-unused-label -Wno-unreachable-code -Wno-unused-variable -Wno-unused-command-line-argument -Wno-unknown-warning-option -Wno-unused-but-set-variable
i686-linux-gnu-gcc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O2 -Wall -g -fstack-protector-strong -Wformat -Werror=format-security -g -fwrapv -O2 -g -O2 -ffile-prefix-map=/<<PKGBUILDDIR>>=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -fPIC -I/<<PKGBUILDDIR>>/mypyc/lib-rt -Ibuild -I/usr/include/python3.9 -c build/misc_ops.c -o build/temp.linux-i686-3.9/build/misc_ops.o -O0 -Werror -Wno-unused-function -Wno-unused-label -Wno-unreachable-code -Wno-unused-variable -Wno-unused-command-line-argument -Wno-unknown-warning-option -Wno-unused-but-set-variable
i686-linux-gnu-gcc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O2 -Wall -g -fstack-protector-strong -Wformat -Werror=format-security -g -fwrapv -O2 -g -O2 -ffile-prefix-map=/<<PKGBUILDDIR>>=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -fPIC -I/<<PKGBUILDDIR>>/mypyc/lib-rt -Ibuild -I/usr/include/python3.9 -c build/set_ops.c -o build/temp.linux-i686-3.9/build/set_ops.o -O0 -Werror -Wno-unused-function -Wno-unused-label -Wno-unreachable-code -Wno-unused-variable -Wno-unused-command-line-argument -Wno-unknown-warning-option -Wno-unused-but-set-variable
i686-linux-gnu-gcc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O2 -Wall -g -fstack-protector-strong -Wformat -Werror=format-security -g -fwrapv -O2 -g -O2 -ffile-prefix-map=/<<PKGBUILDDIR>>=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -fPIC -I/<<PKGBUILDDIR>>/mypyc/lib-rt -Ibuild -I/usr/include/python3.9 -c build/str_ops.c -o build/temp.linux-i686-3.9/build/str_ops.o -O0 -Werror -Wno-unused-function -Wno-unused-label -Wno-unreachable-code -Wno-unused-variable -Wno-unused-command-line-argument -Wno-unknown-warning-option -Wno-unused-but-set-variable
i686-linux-gnu-gcc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O2 -Wall -g -fstack-protector-strong -Wformat -Werror=format-security -g -fwrapv -O2 -g -O2 -ffile-prefix-map=/<<PKGBUILDDIR>>=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -fPIC -I/<<PKGBUILDDIR>>/mypyc/lib-rt -Ibuild -I/usr/include/python3.9 -c build/tuple_ops.c -o build/temp.linux-i686-3.9/build/tuple_ops.o -O0 -Werror -Wno-unused-function -Wno-unused-label -Wno-unreachable-code -Wno-unused-variable -Wno-unused-command-line-argument -Wno-unknown-warning-option -Wno-unused-but-set-variable
creating build/lib.linux-i686-3.9
i686-linux-gnu-gcc -pthread -shared -Wl,-O1 -Wl,-Bsymbolic-functions -Wl,-z,relro -g -fwrapv -O2 -Wl,-z,relro -Wl,-z,now -g -O2 -ffile-prefix-map=/<<PKGBUILDDIR>>=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 build/temp.linux-i686-3.9/build/__native_3ef2a66b8671abc76f6e.o build/temp.linux-i686-3.9/build/__native_native.o build/temp.linux-i686-3.9/build/__native_other.o build/temp.linux-i686-3.9/build/dict_ops.o build/temp.linux-i686-3.9/build/exc_ops.o build/temp.linux-i686-3.9/build/generic_ops.o build/temp.linux-i686-3.9/build/getargs.o build/temp.linux-i686-3.9/build/getargsfast.o build/temp.linux-i686-3.9/build/init.o build/temp.linux-i686-3.9/build/int_ops.o build/temp.linux-i686-3.9/build/list_ops.o build/temp.linux-i686-3.9/build/misc_ops.o build/temp.linux-i686-3.9/build/set_ops.o build/temp.linux-i686-3.9/build/str_ops.o build/temp.linux-i686-3.9/build/tuple_ops.o -o build/lib.linux-i686-3.9/3ef2a66b8671abc76f6e__mypyc.cpython-39-i386-linux-gnu.so
building 'native' extension
i686-linux-gnu-gcc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O2 -Wall -g -fstack-protector-strong -Wformat -Werror=format-security -g -fwrapv -O2 -g -O2 -ffile-prefix-map=/<<PKGBUILDDIR>>=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -fPIC -I/usr/include/python3.9 -c build/native.c -o build/temp.linux-i686-3.9/build/native.o -O0 -Werror -Wno-unused-function -Wno-unused-label -Wno-unreachable-code -Wno-unused-variable -Wno-unused-command-line-argument -Wno-unknown-warning-option -Wno-unused-but-set-variable
i686-linux-gnu-gcc -pthread -shared -Wl,-O1 -Wl,-Bsymbolic-functions -Wl,-z,relro -g -fwrapv -O2 -Wl,-z,relro -Wl,-z,now -g -O2 -ffile-prefix-map=/<<PKGBUILDDIR>>=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 build/temp.linux-i686-3.9/build/native.o -o build/lib.linux-i686-3.9/native.cpython-39-i386-linux-gnu.so
building 'other' extension
i686-linux-gnu-gcc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O2 -Wall -g -fstack-protector-strong -Wformat -Werror=format-security -g -fwrapv -O2 -g -O2 -ffile-prefix-map=/<<PKGBUILDDIR>>=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -fPIC -I/usr/include/python3.9 -c build/other.c -o build/temp.linux-i686-3.9/build/other.o -O0 -Werror -Wno-unused-function -Wno-unused-label -Wno-unreachable-code -Wno-unused-variable -Wno-unused-command-line-argument -Wno-unknown-warning-option -Wno-unused-but-set-variable
i686-linux-gnu-gcc -pthread -shared -Wl,-O1 -Wl,-Bsymbolic-functions -Wl,-z,relro -g -fwrapv -O2 -Wl,-z,relro -Wl,-z,now -g -O2 -ffile-prefix-map=/<<PKGBUILDDIR>>=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 build/temp.linux-i686-3.9/build/other.o -o build/lib.linux-i686-3.9/other.cpython-39-i386-linux-gnu.so
copying build/lib.linux-i686-3.9/3ef2a66b8671abc76f6e__mypyc.cpython-39-i386-linux-gnu.so -> 
copying build/lib.linux-i686-3.9/native.cpython-39-i386-linux-gnu.so -> 
copying build/lib.linux-i686-3.9/other.cpython-39-i386-linux-gnu.so -> 

*** Exit status: 1
----------------------------- Captured stderr call -----------------------------
Expected:
  1
  2
  1                                             (diff)
  2                                             (diff)
Actual:
  1
  2
  Traceback (most recent call last):            (diff)
    File "driver.py", line 8, in <module>       (diff)
      assert b.foo(o) == id(o)                  (diff)
  AssertionError                                (diff)

______________________ testMultiModuleSpecialize_separate ______________________
[gw3] linux -- Python 3.9.7 /usr/bin/python3.9
data: /<<PKGBUILDDIR>>/mypyc/test-data/run-multimodule.test:186:
/<<PKGBUILDDIR>>/mypyc/test/test_run.py:133: in run_case
    self.run_case_inner(testcase)
/<<PKGBUILDDIR>>/mypyc/test/test_run.py:148: in run_case_inner
    self.run_case_step(testcase, step)
/<<PKGBUILDDIR>>/mypyc/test/test_run.py:322: in run_case_step
    assert_test_output(testcase, outlines, msg, expected)
/<<PKGBUILDDIR>>/mypyc/test/testutil.py:163: in assert_test_output
    assert_string_arrays_equal(
/<<PKGBUILDDIR>>/mypy/test/helpers.py:117: in assert_string_arrays_equal
    raise AssertionError(msg)
E   AssertionError: Invalid output (/<<PKGBUILDDIR>>/mypyc/test-data/run-multimodule.test, line 186)
----------------------------- Captured stdout call -----------------------------
running build_ext
building 'native__mypyc' extension
creating build/temp.linux-i686-3.9
creating build/temp.linux-i686-3.9/build
i686-linux-gnu-gcc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O2 -Wall -g -fstack-protector-strong -Wformat -Werror=format-security -g -fwrapv -O2 -g -O2 -ffile-prefix-map=/<<PKGBUILDDIR>>=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -fPIC -I/<<PKGBUILDDIR>>/mypyc/lib-rt -Ibuild -I/usr/include/python3.9 -c build/__native_native.c -o build/temp.linux-i686-3.9/build/__native_native.o -O0 -Werror -Wno-unused-function -Wno-unused-label -Wno-unreachable-code -Wno-unused-variable -Wno-unused-command-line-argument -Wno-unknown-warning-option -Wno-unused-but-set-variable
creating build/lib.linux-i686-3.9
i686-linux-gnu-gcc -pthread -shared -Wl,-O1 -Wl,-Bsymbolic-functions -Wl,-z,relro -g -fwrapv -O2 -Wl,-z,relro -Wl,-z,now -g -O2 -ffile-prefix-map=/<<PKGBUILDDIR>>=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 build/temp.linux-i686-3.9/build/__native_native.o -o build/lib.linux-i686-3.9/native__mypyc.cpython-39-i386-linux-gnu.so
building 'native' extension
i686-linux-gnu-gcc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O2 -Wall -g -fstack-protector-strong -Wformat -Werror=format-security -g -fwrapv -O2 -g -O2 -ffile-prefix-map=/<<PKGBUILDDIR>>=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -fPIC -I/usr/include/python3.9 -c build/native.c -o build/temp.linux-i686-3.9/build/native.o -O0 -Werror -Wno-unused-function -Wno-unused-label -Wno-unreachable-code -Wno-unused-variable -Wno-unused-command-line-argument -Wno-unknown-warning-option -Wno-unused-but-set-variable
i686-linux-gnu-gcc -pthread -shared -Wl,-O1 -Wl,-Bsymbolic-functions -Wl,-z,relro -g -fwrapv -O2 -Wl,-z,relro -Wl,-z,now -g -O2 -ffile-prefix-map=/<<PKGBUILDDIR>>=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 build/temp.linux-i686-3.9/build/native.o -o build/lib.linux-i686-3.9/native.cpython-39-i386-linux-gnu.so
building 'other__mypyc' extension
i686-linux-gnu-gcc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O2 -Wall -g -fstack-protector-strong -Wformat -Werror=format-security -g -fwrapv -O2 -g -O2 -ffile-prefix-map=/<<PKGBUILDDIR>>=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -fPIC -I/<<PKGBUILDDIR>>/mypyc/lib-rt -Ibuild -I/usr/include/python3.9 -c build/__native_other.c -o build/temp.linux-i686-3.9/build/__native_other.o -O0 -Werror -Wno-unused-function -Wno-unused-label -Wno-unreachable-code -Wno-unused-variable -Wno-unused-command-line-argument -Wno-unknown-warning-option -Wno-unused-but-set-variable
i686-linux-gnu-gcc -pthread -shared -Wl,-O1 -Wl,-Bsymbolic-functions -Wl,-z,relro -g -fwrapv -O2 -Wl,-z,relro -Wl,-z,now -g -O2 -ffile-prefix-map=/<<PKGBUILDDIR>>=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 build/temp.linux-i686-3.9/build/__native_other.o -o build/lib.linux-i686-3.9/other__mypyc.cpython-39-i386-linux-gnu.so
building 'other' extension
i686-linux-gnu-gcc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O2 -Wall -g -fstack-protector-strong -Wformat -Werror=format-security -g -fwrapv -O2 -g -O2 -ffile-prefix-map=/<<PKGBUILDDIR>>=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -fPIC -I/usr/include/python3.9 -c build/other.c -o build/temp.linux-i686-3.9/build/other.o -O0 -Werror -Wno-unused-function -Wno-unused-label -Wno-unreachable-code -Wno-unused-variable -Wno-unused-command-line-argument -Wno-unknown-warning-option -Wno-unused-but-set-variable
i686-linux-gnu-gcc -pthread -shared -Wl,-O1 -Wl,-Bsymbolic-functions -Wl,-z,relro -g -fwrapv -O2 -Wl,-z,relro -Wl,-z,now -g -O2 -ffile-prefix-map=/<<PKGBUILDDIR>>=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 build/temp.linux-i686-3.9/build/other.o -o build/lib.linux-i686-3.9/other.cpython-39-i386-linux-gnu.so
copying build/lib.linux-i686-3.9/native__mypyc.cpython-39-i386-linux-gnu.so -> 
copying build/lib.linux-i686-3.9/native.cpython-39-i386-linux-gnu.so -> 
copying build/lib.linux-i686-3.9/other__mypyc.cpython-39-i386-linux-gnu.so -> 
copying build/lib.linux-i686-3.9/other.cpython-39-i386-linux-gnu.so -> 

*** Exit status: 1
----------------------------- Captured stderr call -----------------------------
Expected:
  1
  2
  1                                             (diff)
  2                                             (diff)
Actual:
  1
  2
  Traceback (most recent call last):            (diff)
    File "driver.py", line 8, in <module>       (diff)
      assert b.foo(o) == id(o)                  (diff)
  AssertionError                                (diff)

=============================== warnings summary ===============================
mypyc/test/test_run.py::TestRun::testFinalStaticRunListTupleInt
mypyc/test/test_run.py::TestRun::testAsync
mypyc/test/test_run.py::TestRun::testClassVariable
  /usr/lib/python3/dist-packages/setuptools/lib2to3_ex.py:10: PendingDeprecationWarning: lib2to3 package is deprecated and may not be able to parse Python 3.10+
    from lib2to3.refactor import RefactoringTool, get_fixers_from_package

-- Docs: https://docs.pytest.org/en/stable/warnings.html
=========================== short test summary info ============================
FAILED mypyc/test/test_run.py::TestRun::testSubclassSpecialize1
FAILED mypyc/test/test_run.py::TestRun::testSubclassSpecialize2
FAILED mypyc/test/test_run.py::TestRun::testMultiModuleSpecialize
FAILED mypyc/test/test_run.py::TestRunMultiFile::testMultiModuleSpecialize_multi
FAILED mypyc/test/test_run.py::TestRunSeparate::testMultiModuleSpecialize_separate
===== 5 failed, 9170 passed, 369 skipped, 3 warnings in 552.24s (0:09:12) ======
```
https://buildd.debian.org/status/fetch.php?pkg=mypy&arch=i386&ver=0.910-1&stamp=1632141527&raw=0

**Your Environment**

- Mypy version used: 0.910
- Mypy command-line flags: see above
- Mypy configuration options from `mypy.ini` (and other config files): as distributed in mypy 0.910
- Python version used: 3.9.7
- Operating system and version: Debian's developmement distribution "unstable"

I appear to see the same failures on HPPA:

```
*** Exit status: 1
----------------------------- Captured stderr call -----------------------------
Expected:
  1
  2
  1                                             (diff)
  2                                             (diff)
Actual:
  1
  2
  Traceback (most recent call last):            (diff)
    File "driver.py", line 8, in <module>       (diff)
      assert b.foo(o) == id(o)                  (diff)
  AssertionError                                (diff)

______________________ testMultiModuleSpecialize_separate ______________________
[gw1] linux -- Python 3.9.7 /usr/bin/python3.9
data: /var/tmp/portage/dev-python/mypy-0.910-r1/work/mypy-0.910/mypyc/test-data/run-multimodule.test:186:
../../work/mypy-0.910/mypyc/test/test_run.py:133: in run_case
    self.run_case_inner(testcase)
../../work/mypy-0.910/mypyc/test/test_run.py:148: in run_case_inner
    self.run_case_step(testcase, step)
../../work/mypy-0.910/mypyc/test/test_run.py:322: in run_case_step
    assert_test_output(testcase, outlines, msg, expected)
../../work/mypy-0.910/mypyc/test/testutil.py:163: in assert_test_output
    assert_string_arrays_equal(
../../work/mypy-0.910/mypy/test/helpers.py:117: in assert_string_arrays_equal
    raise AssertionError(msg)
E   AssertionError: Invalid output (/var/tmp/portage/dev-python/mypy-0.910-r1/work/mypy-0.910/mypyc/test-data/run-multimodule.test, line 186)
----------------------------- Captured stdout call -----------------------------
``

```
FAILED mypyc/test/test_run.py::TestRun::testSubclassSpecialize1
FAILED mypyc/test/test_run.py::TestRun::testSubclassSpecialize2
FAILED mypyc/test/test_run.py::TestRun::testMultiModuleSpecialize
FAILED mypyc/test/test_run.py::TestRunMultiFile::testMultiModuleSpecialize_multi
FAILED mypyc/test/test_run.py::TestRunSeparate::testMultiModuleSpecialize_separate
```

--- a/mypyc/lib-rt/CPy.h
+++ b/mypyc/lib-rt/CPy.h
@@ -121,6 +121,7 @@ static inline size_t CPy_FindAttrOffset(PyTypeObject *trait, CPyVTableItem *vtab
 
 
 CPyTagged CPyTagged_FromSsize_t(Py_ssize_t value);
+CPyTagged CPyTagged_FromVoidPtr(void *ptr);
 CPyTagged CPyTagged_FromObject(PyObject *object);
 CPyTagged CPyTagged_StealFromObject(PyObject *object);
 CPyTagged CPyTagged_BorrowFromObject(PyObject *object);

--- a/mypyc/lib-rt/int_ops.c
+++ b/mypyc/lib-rt/int_ops.c
@@ -26,6 +26,15 @@ CPyTagged CPyTagged_FromSsize_t(Py_ssize_t value) {
     }
 }
 
+CPyTagged CPyTagged_FromVoidPtr(void *ptr) {
+    if ((uintptr_t)ptr > PY_SSIZE_T_MAX) {
+        PyObject *object = PyLong_FromVoidPtr(ptr);
+        return ((CPyTagged)object) | CPY_INT_TAG;
+    } else {
+        return CPyTagged_FromSsize_t((Py_ssize_t)ptr);
+    }
+}
+
 CPyTagged CPyTagged_FromObject(PyObject *object) {
     int overflow;
     // The overflow check knows about CPyTagged's width

--- a/mypyc/lib-rt/misc_ops.c
+++ b/mypyc/lib-rt/misc_ops.c
@@ -437,7 +437,7 @@ fail:
 }
 
 CPyTagged CPyTagged_Id(PyObject *o) {
-    return CPyTagged_FromSsize_t((Py_ssize_t)o);
+    return CPyTagged_FromVoidPtr(o);
 }
 
 #define MAX_INT_CHARS 22

